ddev/tests/cli/meta/scripts/test_upgrade_python.py

```diff
@@ -3,39 +3,42 @@
 # Licensed under a 3-clause BSD style license (see LICENSE)
 from collections import defaultdict
 
-from ddev.repo.constants import PYTHON_VERSION
 
-
-def test_upgrade_python(ddev, repository):
-    major, minor = PYTHON_VERSION.split('.')
-    new_version = f'{major}.{int(minor) + 1}'
+def test_upgrade_python(fake_repo, ddev):
+    new_version = "3.11"
+    old_version = "3.9"
 
     changes = defaultdict(list)
-    for entry in repository.path.iterdir():
+    for entry in fake_repo.path.iterdir():
         config_file = entry / 'hatch.toml'
         if not config_file.is_file():
             continue
 
         for i, line in enumerate(config_file.read_text().splitlines()):
-            if line.startswith('python = [') and PYTHON_VERSION in line:
+            if line.startswith('python = [') and old_version in line:
                 changes[config_file].append(i)
 
     minimum_changes = sum(map(len, changes.values()))
```

Comment:
It might be worth dropping this loop? IIUC it's only used to make an assertion on the number after `Passed:` in the output, and since the output is deterministic now we might be able to drop that (we're already asserting that it equals 2 now).

Reply:
Good catch! Updated